### PR TITLE
Run streams resetter for external source topics and intermediate topics

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: src/main/java/com/bakdata/common_kafka_streams/util/TopologyInformation.java [hunk 1]
```diff
@@ -42,6 +43,37 @@ public TopologyInformation(final Topology topology, final String streamsId) {
...
+    private static Stream<String> getAllSinks(final Collection<Node> nodes) {
+        return nodes.stream()
+                .filter(node -> node instanceof Sink)
+                .map(node -> ((Sink) node))
+                .map(Sink::topic);
+    }
+
+    private static boolean isInternalTopic(final String topic) {
+        return topic.startsWith("KSTREAM-") || topic.startsWith("KTABLE-")
+                || topic.endsWith("-repartition") || topic.endsWith("-changelog");
```
Comment (torbsto) [inline src/main/java/com/bakdata/common_kafka_streams/util/TopologyInformation.java]: assuming all topics with those suffixes being internal doesn't seem right. 
For `"-changelog",` you can check if it is a backing topic (i.e. `getBackingTopics(nodes).contains(topic)`). `"-repartition"` would require an analogous method to `getBackingTopics`.

Comment (philipp94831) [reply src/main/java/com/bakdata/common_kafka_streams/util/TopologyInformation.java]: Hm. In what case do we consider a topic ending with -repartition not internal?

Comment (torbsto) [reply src/main/java/com/bakdata/common_kafka_streams/util/TopologyInformation.java]: Depending on the definition of internal, I would argue that topics used in `through` are not internal. But obviously they can end with -repartition.

Comment (philipp94831) [reply src/main/java/com/bakdata/common_kafka_streams/util/TopologyInformation.java]: What do you think now?

**Pull request merged**